app/src/main/java/com/example/pocketwatching/Activities/ProfileActivity.java

```diff
@@ -81,36 +97,137 @@ public void onClick(View v) {
         });
     }
 
-    private void goMainActivity() {
-        Intent i = new Intent(this, MainActivity.class);
-        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP); // this makes sure the Back button won't work
-        i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK); // same as above
-        startActivity(i);
-        finish();
-    }
-
+    // gets EthWallet object from given address
     private synchronized void getEthWallet(String address) {
         Call<EthWallet> call = (Call<EthWallet>) EthplorerClient.getInstance().getEthplorerApi().getEthWallet(address);
         call.enqueue(new Callback<EthWallet>() {
             @Override
             public void onResponse(Call<EthWallet> call, Response<EthWallet> response) {
                 userEthWallets.add(response.body());
                 if (userEthWallets.size() == userWallets.size()) {
+                    initValuableTokens();
                     populateProfile();
                 }
             }
 
             @Override
             public void onFailure(Call<EthWallet> call, Throwable t) {
+                // how much does line length matter?
```

Comment:
Is this referring to ```Toast.LENGTH_SHORT```? If so, that refers to the duration of time that the Toast will be visible before it's dismissed. So LENGTH_SHORT means only show the Toast for a short time before it hides itself. LENGTH_LONG shows the Toast for a longer time. Keep in mind that these values are flags and not actual time durations in ms (see https://stackoverflow.com/a/7965192)

Reply:
I was referring to the actual length of the line in the code